rasa/utils/tensorflow/models.py

```diff
@@ -820,11 +826,31 @@ def _get_mask_for(
     def _get_sequence_feature_lengths(
         self, tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]], key: Text
     ) -> tf.Tensor:
-        """Fetches the sequence lengths of real tokens per input example.
+        """Tries to create sequence length information for the given key.
+
+        Note that, if no explicit sequence length information is present in the batch,
+        this function assumes the respective sequence-level features are all empty.
+        This behaviour is useful in case we deal with empty sequences only (e.g.
+        DIET without transformer layers and no need for sequential output).
+
+        Remember, the number of real tokens for an example is the same as the length
+        of the sequence of the sequence-level (token-level) features for that input
+        example.
+
+        Returns:
+           a tensor of shape (batch_size, ), always
 
-        The number of real tokens for an example is the same as the length of the
-        sequence of the sequence-level (token-level) features for that input example.
+        Raises:
+           a RasaModelConfigException in case no SEQUENCE subkey can be found for the
+           given key
         """
+        if SEQUENCE not in tf_batch_data[key]:
```

Comment:
Shouldn't this `if` be placed in between the `if` on line 854 because `key` might not be present in `tf_batch_data` and lookup `tf_batch_data[key]` will fail. Alternatively, you could do a `.get()`

Reply:
Oh yep, I've build a potential bug there! Just (should) also just pass because of the defaultdicts 😬

Reply:
fixed